GitUI/BranchTreePanel/RepoObjectsTree.Nodes.Submodules.cs

```diff
@@ -260,23 +263,67 @@ private void OnStatusUpdated(SubmoduleStatusEventArgs e)
                 {
                     CancellationTokenSource? cts = null;
                     Task<Nodes>? loadNodesTask = null;
-                    await ReloadNodesAsync(token =>
+                    if (e.Structure)
                     {
-                        cts = CancellationTokenSource.CreateLinkedTokenSource(e.Token, token);
-                        loadNodesTask = LoadNodesAsync(e.Info, cts.Token);
-                        return loadNodesTask;
-                    }).ConfigureAwait(false);
+                        _currentNodes = null;
+                    }
+
+                    if (_currentNodes is not null)
+                    {
+                        // Structure is up-to-date, update status
+                        var infos = e.Info.AllSubmodules.ToDictionary(info => info.Path, info => info);
+                        Validates.NotNull(e.Info.TopProject);
+                        infos[e.Info.TopProject.Path] = e.Info.TopProject;
+                        var nodes = _currentNodes.DepthEnumerator<SubmoduleNode>().ToList();
+                        foreach (var node in nodes)
+                        {
+                            if (infos.ContainsKey(node.Info.Path))
+                            {
+                                node.Info = infos[node.Info.Path];
```

Comment:
An option here is to check if the status has changed and only update the required nodes.
(The tooltip requires an additional git command, so there is some benefit).
Will increase the complexity though.